client/src/Classifier.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef classifier_h
+#define classifier_h
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/ml/ml.hpp>
+
+using namespace std;
+using namespace cv;
+
+class Classifier {
+  CvSVM svm;
+
+public:
+  Classifier(string filename) {
+    svm.load(filename.c_str());
+  }
+
+  // Guess the characters name from the image icon.
+  string classify(const Mat &image) {
```

Comment:
I had hoped you were just blissfully ignorant of that argument and
therefore could be bent to my will easily

On Wed, Jun 1, 2016 at 2:20 PM Guntars Ašmanis <notifications@github.com>
wrote:

> In client/src/Classifier.hpp
> <https://github.com/Pathgather/smashgather/pull/2#discussion_r65415086>:
>
> > +#include <opencv2/imgproc/imgproc.hpp>
> > +#include <opencv2/ml/ml.hpp>
> > +
> > +using namespace std;
> > +using namespace cv;
> > +
> > +class Classifier {
> > +  CvSVM svm;
> > +
> > +public:
> > +  Classifier(string filename) {
> > +    svm.load(filename.c_str());
> > +  }
> > +
> > +  // Guess the characters name from the image icon.
> > +  string classify(const Mat &image) {
>
> The argument about *Is pointer / reference part of the type or a part of
> the binding* is probably as old as tabs v. spaces. But fine, for
> consistency, let's make it Mat& image.
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/Pathgather/smashgather/pull/2/files/2ca41645b57a77b6cdd1d87c793b0d071cc7fd4c..d785e2ae34dfc15f0922ea2cf8041453d7d2c9a2#r65415086>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABv9N1hz3GY9mae8EhB2hUwlscv7yf7Vks5qHc1agaJpZM4Ipbpc>
> .
>